src/ceph-volume/ceph_volume/util/device.py

```diff
@@ -99,6 +102,26 @@ def __init__(self, path):
 
         self.device_id = self._get_device_id()
 
+    def fetch_lsm(self):
+        '''
+        Attempt to fetch libstoragemgmt (LSM) metadata, and return to the caller
+        as a dict. An empty dict is passed back to the caller if the target path
+        is not a block device, or lsm is unavailable on the host. Otherwise the
+        json returned will provide LSM attributes, and any associated errors that
+        lsm encountered when probing the device.
+        '''
+        devName = self.path.split('/')[-1]
+        if not os.path.exists('/sys/block/{}'.format(devName)):
+            return {}
+
+        lsm_disk = LSMDisk(self.path)
+        if not lsm_disk.lsm_available:
```

Comment:
it's more readable from my perspective. You instantiate LSMDisk, and check if it has lsm_available before you do anything - rather than instantiate and rely on the report method

Reply:
will change